### PR TITLE
#2 Client feature 2 signup and login style overhaul

### PR DESCRIPTION
**Pull request by jonahtabb**
### What this PR does:
- Addresses issue #2 
- Major style overhaul to make authorization components match mock-ups.
- Includes major changes to the following components on the Front End:
  - Login, LoginForm, Signup, SignupForm, AuthHeader
- Refactored some common styles to use overrides in the material UI theme provider (theme.ts)
 - This required the use of theme augmentation and installation of Maertial Ui Lab
- Removed username field (based on mock), and adjusted associated front and and backend
- Improvements to mobile responsiveness include:
 - Hamburger menu on tablet and smaller
 - Some additional breakpoint styling

### How to test this PR:
- Fill out and submit sign up form
- Fill out and submit login form
_Both these actions should provide access to the dashboard_

### Outstanding Issues:
- Failing tests that need to be updated.  Mainly due to text changes in the UI / design that were done to match the mock.
- Buttons on mobile nav menu are not functional
- Site automatically attempts to authorize with the server on every page load with a useEffect in UseAuthContext .  I'm not sure if this is expected behavior, or if there is a way to only ping this route if a cookie / data exists.
- Using Roboto as the default font, although the mock font seems different
- Unnecessary commented code in AuthHeader still needs to be removed.

### Screenshots / Videos:
- Loom Video
 <a href="https://www.loom.com/share/e5c601fa907543429d1ec6f031578763"  rel="external" target="_blank">Foo</a>

- Screenshots
 image
 image
 image

### CODE REVIEW

File: client/src/components/AuthHeader/AuthHeader.tsx [hunk 1]
```diff
@@ -1,27 +1,111 @@
...
+                <Button color="primary" variant="contained">
+                  {signupButtonText}
+                </Button>
+              </Link>
+            </Box>
+          )}
+        </Toolbar>
+      </AppBar>
     </Box>
+    // <Box p={1} className={classes.authHeader}>
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/AuthHeader.tsx:101]: Please remove commented code or add a relevant TODO comment explaining the reason for commenting this out

File: client/src/components/AuthHeader/AuthHeader.tsx [hunk 2]
```diff
@@ -1,27 +1,111 @@
...
+              >
+                <MenuItem onClick={handleMenuClose}>{loginButtonText}</MenuItem>
+                <MenuItem onClick={handleMenuClose}>{signupButtonText}</MenuItem>
+              </Menu>
+            </>
+          ) : (
+            <Box className={classes.authLinks}>
+              <Typography className={classes.accAside}>{asideText}</Typography>
+              <Link to={loginButtonLinkTo} className={classes.link}>
+                <Button color="primary" variant="outlined">
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/AuthHeader.tsx:87]:
```suggestion
                <Button component={Link} to={loginButtonLinkTo} color="primary" variant="outlined">
```
The MUI Button and react-router's Link can be combined :)
Ditto throughout!

File: client/src/components/AuthHeader/useStyles.ts [hunk 3]
```diff
@@ -7,6 +7,17 @@ const useStyles = makeStyles((theme) => ({
     alignItems: 'center',
     display: 'flex',
     flexWrap: 'wrap',
+    // flexGrow: 1,
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/useStyles.ts:10]: Please remove

File: client/src/pages/SignUp/useStyles.ts [hunk 4]
```diff
@@ -13,13 +13,27 @@ const useStyles = makeStyles(() => ({
     justifyContent: 'space-between',
     flexDirection: 'column',
     minHeight: '100vh',
-    paddingTop: 23,
+  },
+  signupWrapper: {
+    display: 'flex',
```
Comment (rajivtitus) [inline client/src/pages/SignUp/useStyles.ts:18]: Some of these significant styles can be added to the Box component directly.

Eg: `<Box display="flex" alignItems="center"...>`

File: client/src/pages/Login/useStyles.ts [hunk 5]
```diff
@@ -13,13 +13,27 @@ const useStyles = makeStyles(() => ({
     justifyContent: 'space-between',
     flexDirection: 'column',
     minHeight: '100vh',
-    paddingTop: 23,
+  },
+  loginWrapper: {
+    display: 'flex',
```
Comment (rajivtitus) [inline client/src/pages/Login/useStyles.ts:18]: Some of these significant styles can be added to the Box component directly.

Eg: `<Box display="flex" alignItems="center"...>`

**Review by rajivtitus — COMMENTED** [review]
Great work with this! 
- Really good use of MUI
- Nice touch adding mobile responsiveness

Just a few minor comments, please have a look and address. Thanks!

File: client/src/components/AuthHeader/AuthHeader.tsx (hunk 1 above)
Comment (jonahtabb) [reply client/src/components/AuthHeader/AuthHeader.tsx]: Resloved by a0e5bffa1dde60ac01d6caee19d3a1150fbf63ec

File: client/src/components/AuthHeader/AuthHeader.tsx (hunk 2 above)
Comment (jonahtabb) [reply client/src/components/AuthHeader/AuthHeader.tsx]: Resolved by a4912e06cc41cb7e511c0ab8b8d2c0fe5077f483

File: client/src/pages/SignUp/useStyles.ts (hunk 4 above)
Comment (jonahtabb) [reply client/src/pages/SignUp/useStyles.ts]: Resolved by bb371d73a565f65dd3325c167df7ea148f9d2c81

File: client/src/components/AuthHeader/useStyles.ts (hunk 3 above)
Comment (jonahtabb) [reply client/src/components/AuthHeader/useStyles.ts]: removed

File: client/src/pages/Login/useStyles.ts (hunk 5 above)
Comment (jonahtabb) [reply client/src/pages/Login/useStyles.ts]: Resolved by bb371d73a565f65dd3325c167df7ea148f9d2c81

**Pull request merged**